backend/services/logging_service.py

```diff
@@ -28,9 +28,12 @@ def log_query(user_id: str, raw_text: str, enhanced_text: str = None):
         raise
 
 
-def log_interaction(user_id: str, query_id: str, clicked_url: str, rank: int):
+def log_interaction(user_id: str, query_id: str, clicked_url: str, rank: int, action_type: str = "click"):
```

Comment:
action_type: str = "click" is never used here. 
```suggestion
def log_interaction(user_id: str, query_id: str, clicked_url: str, rank: int):
```